src/domain/chains/chains.repository.spec.ts

```diff
@@ -0,0 +1,302 @@
+ * Whilst `getAllChains` is to some extent as well, the following
+ * tests are required to cover all edge cases.
+ */
+describe('ChainsRepository', () => {
+  // According to the limits of the Config Service
+  // @see https://github.com/safe-global/safe-config-service/blob/main/src/chains/views.py#L14-L16
+  const OFFSET = 40;
+
+  let target: ChainsRepository;
+  const maxSequentialPages = 3;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    mockConfigurationService.getOrThrow.mockImplementation((key) => {
+      if (key === 'safeConfig.chains.maxSequentialPages')
+        return maxSequentialPages;
+    });
+
+    target = new ChainsRepository(
+      mockLoggingService,
+      mockConfigApi,
+      mockTransactionApiManager,
+      mockConfigurationService,
+    );
+  });
+
+  it('should return all chains across pages below request limit', async () => {
+    const url = faker.internet.url({ appendSlash: true });
+    const chains = Array.from(
+      {
+        length: ChainsRepository.MAX_LIMIT * (maxSequentialPages - 1), // One page less than request limit
+      },
+      (_, i) => chainBuilder().with('chainId', i.toString()).build(),
+    );
+    const pages = chunk(chains, ChainsRepository.MAX_LIMIT).map(
+      (results, i, arr) => {
+        const pageOffset = (i + 1) * OFFSET;
+
+        const previous = ((): string | null => {
+          if (i === 0) {
+            return null;
+          }
+          return limitAndOffsetUrlFactory(
+            ChainsRepository.MAX_LIMIT,
+            pageOffset,
+            url,
+          );
+        })();
+        const next = ((): string | null => {
+          if (i === arr.length - 1) {
+            return null;
+          }
+          return limitAndOffsetUrlFactory(
+            ChainsRepository.MAX_LIMIT,
+            pageOffset,
+            url,
+          );
+        })();
+
+        return pageBuilder<Chain>()
+          .with('results', results)
+          .with('count', chains.length)
+          .with('previous', previous)
+          .with('next', next)
+          .build();
+      },
+    );
+    mockConfigApi.getChains.mockImplementation(
+      ({ offset }): Promise<Page<Chain>> => {
+        if (offset === 0) {
+          return Promise.resolve(pages[0]);
+        }
+        if (offset === 40) {
+          return Promise.resolve(pages[1]);
+        }
+        if (offset === 80) {
+          return Promise.resolve(pages[2]);
+        }
+        return Promise.reject(new Error('Invalid offset'));
+      },
+    );
+
+    const result = await target.getAllChains();
+
+    expect(result).toStrictEqual(
+      chains.map((chain) => {
+        return {
+          ...chain,
+          ensRegistryAddress: getAddress(chain.ensRegistryAddress!),
+        };
+      }),
+    );
+    expect(mockConfigApi.getChains).toHaveBeenCalledTimes(2);
+    expect(mockConfigApi.getChains).toHaveBeenNthCalledWith(1, {
+      limit: 40,
```

Comment:
Can we use `MAX_LIMIT` instead of 40?

Reply:
Changed to use constant in 3b4b46c25b4d282d29daa09f485730ab65e2dc65.